### PR TITLE
Converting some Blink tests over & tightening upgrade logic

### PR DESCRIPTION
**Pull request by blois**
I'm working on using the Polymer polyfills for Dart when compiled to JS and we're using a number of tests from Blink. These tests picked up a number of differences which appear straightforward enough to be covered by the polyfill.

Not quite sure what level of adherence to the spec the goal is for the polyfill, many of the issues were around negative testing.

These changes are primarily to get the polyfill to pass:
https://code.google.com/p/chromium/codesearch#chromium/src/third_party/WebKit/LayoutTests/fast/dom/custom/document-register-type-extensions.html

Though I left off the createElementNS tests.

### CODE REVIEW

File: src/CustomElements.js [hunk 1]
```diff
@@ -309,20 +312,43 @@ if (useNative) {
...
+
+    if (typeExtension) {
+      var element = createElement(tag);
+      element.setAttribute('is', typeExtension);
+      return element;
+    }
+    var element = domCreateElement(tag);
+    // Custom tags should be HTMLElements even if not upgraded.
+    if (tag.indexOf('-') >= 0) {
+      implement(element, {
```
Comment (arv) [inline src/CustomElements.js]: The following will do the same:

implement(element, HTMLElement)

File: tools [hunk 2]
```diff
@@ -1 +1 @@
-Subproject commit 973cb8b063100f726622509536cb5114e0300332
```
Comment (arv) [inline tools]: Did you accidentally update this?

**Comment by arv** [comment]
Looks like a good patch

**Comment by sjmiles** [comment]
Thank you for the PR. These are core modules, so we need to do some more review and then will get back to you.

**Comment by blois** [comment]
Fully understand, no specific urgency.

**Comment by jmesserly** [comment]
friendly ping :)

I just hit this when I tried to upgrade the Pub custom_elements package. Would be nice if we can get our code unforked :)

Pete hit this because he ported Blink tests... we're testing that the polyfill matches native, that's how we hit these cases.

**Comment by jmesserly** [comment]
I rebased+squashed this: https://github.com/Polymer/CustomElements/pull/97

**Pull request closed without merging**